### PR TITLE
Fix aes(groupS)

### PR DESCRIPTION
**Pull request by tdhock**
hi this is my proposed fix for #4072, along with a test that fails using current master.

### CODE REVIEW

File: tests/testthat/test-aes-grouping.r [hunk 1]
```diff
@@ -23,6 +23,12 @@ test_that("one group per combination of discrete vars", {
   expect_equal(groups(plot), 4)
 })
 
+test_that("no error for aes(groupS)", {
+  plot <- ggplot(df, aes(a, b, groupS=1)) + geom_point()
+  g <- ggplotGrob(plot)
+  expect_is(g, "grob")
```
Comment (yutannihilation) [inline tests/testthat/test-aes-grouping.r:29]: Could you test `add_group()` directly with `expect_error(..., NULL)`? Using an unknown aes is not a guaranteed usage (c.f. https://github.com/tidyverse/ggplot2/issues/3725#issuecomment-573716059), so you need to create a custom Geom that actually requires `groupS`, but I think it's a bit too heavy to test this issue.

Comment (yutannihilation) [reply tests/testthat/test-aes-grouping.r:29]: If you need some help, I'd be happy to provide the test code.

**Comment by yutannihilation** [comment]
@tdhock Could you fix the test case?

**Comment by tdhock** [comment]
sorry I have lots of other deadlines coming up right now so I won't be able
to fix right away.

**Comment by yutannihilation** [comment]
I see. Sorry to hurry you, maybe I can take care of the test.

**Comment by tdhock** [comment]
great

**Pull request merged**